packages/react-static-tweets/src/twitter.tsx

```diff
@@ -0,0 +1,63 @@
+import React, { Consumer, createContext, Provider, useContext } from 'react'
+import { ConfigInterface } from 'swr'
+
+// TODO: make this more specific
+export type TweetAst = Array<any>
+
+export interface TwitterContext {
+  // static tweet ast info
+  tweetAstMap: TweetAstMap
+
+  // SWR config for dynamically fetching tweet ast info
+  swrOptions: ConfigInterface
+}
+
+export interface TweetAstMap {
+  [tweetId: string]: TweetAst
+}
+
+// Saves the tweets returned as props to the page
+const OriginalTwitter = createContext<TwitterContext>({
+  tweetAstMap: {},
+  swrOptions: {
+    fetcher: (id) =>
+      fetch(`https://twitter-search.vercel.app/api/get-tweet-ast/${id}`).then(
+        (r) => r.json()
+      )
+  }
+})
+
+export function useTwitter () {
+  const twitterContext = useContext(OriginalTwitter)
+
+  return twitterContext
+}
+
+type OverridingProvider<T> = Provider<Partial<T>>
+
+type OverridableContext<T> = {
+  Provider: OverridingProvider<T>,
+  Consumer: Consumer<T>,
+  displayName?: string
+}
+
+// TODO: this is more correct, but maybe it is too verbose?
+type TwitterProviderProps = Parameters<OverridableContext<TwitterContext>['Provider']>[0]
+
+// allows partials that override outer providers
+function OverridingTwitterProvider ({ value, children }: TwitterProviderProps) {
```

Comment:
i understand that you're trying to do the overridable context in a generic way, but i think it would be cleaner if you just solved the problem for this one instance instead of doing it generically. it would remove the `OverridableContext` and `$$typeof` stuff that isn't really related to this library.

Reply:
Yep, that's what I was thinking. Overridable contexts seem useful though. Another lin idea for the to do list! Will refactor soon.

Reply:
yeah; this could totally be its own, reusable standalone react lib.

maybe start off w/ a gist? 👀